buidl/mnemonic.py

```diff
@@ -26,2090 +30,86 @@ def secure_mnemonic(num_bits=256, extra_entropy=0):
+    num_words = len(words)
+    for word in words:
+        all_bits <<= 11
+        all_bits += BIP39.index(word)
+    num_checksum_bits = num_words // 3
+    checksum = all_bits & ((1 << num_checksum_bits) - 1)
+    all_bits >>= num_checksum_bits
+    num_bytes = (num_words * 11 - num_checksum_bits) // 8
+    s = int_to_big_endian(all_bits, num_bytes)
+    computed_checksum = sha256(s)[0] >> (8 - num_checksum_bits)
+    if checksum != computed_checksum:
+        raise ValueError("Checksum is wrong")
+    return s
+
+
+def bytes_to_mnemonic(b, num_bits):
+    """returns a mnemonic given a byte representation"""
+    preseed = big_endian_to_int(b)
     # 1 extra bit for checksum is needed per 32 bits
-    checksum_bits_needed = num_bits // 32
+    num_checksum_bits = num_bits // 32
     # the checksum is the sha256's first n bits. At most this is 8
-    checksum = sha256(s)[0] >> (8 - checksum_bits_needed)
+    checksum = sha256(b)[0] >> (8 - num_checksum_bits)
     # we concatenate the checksum to the preseed
-    total = (preseed << checksum_bits_needed) | checksum
+    all_bits = (preseed << num_checksum_bits) | checksum
     # now we get the mnemonic passphrase
     mnemonic = []
     # now group into groups of 11 bits
-    for _ in range((num_bits + checksum_bits_needed) // 11):
+    for _ in range((num_bits + num_checksum_bits) // 11):
         # grab the last 11 bits
-        current = total & ((1 << 11) - 1)
+        current = all_bits & ((1 << 11) - 1)
         # insert the correct word at the front
-        mnemonic.insert(0, WORD_LIST[current])
+        mnemonic.insert(0, BIP39.word(current))
         # shift by 11 bits so we can move to the next set
-        total >>= 11
+        all_bits >>= 11
     # return the mnemonic phrase by putting spaces between
     return " ".join(mnemonic)
 
 
-WORD_LOOKUP = {}
+class WordList:
+    @classmethod
+    def _load(cls):
+        word_file = path.join(path.dirname(__file__), cls.filename)
+        with open(word_file, "r") as f:
+            cls.words = f.read().split()
+        cls.lookup = {}
+        for i, word in enumerate(cls.words):
+            # add the word's index in the dict lookup
+            cls.lookup[word] = i
+            # if the word is more than 4 characters, also keep
+            #  a lookup of just the first 4 characters
+            if len(word) > 4:
+                cls.lookup[word[:4]] = i
+
+    @classmethod
+    def index(cls, word):
+        return cls.lookup[word]
 
+    @classmethod
+    def normalize(cls, word):
+        return cls.words[cls.lookup[word]]
 
-def __init__():
-    # go through every word
-    for i, word in enumerate(WORD_LIST):
-        # add the word's index in the hash WORD_LOOKUP
-        WORD_LOOKUP[word] = i
-        # if the word is more than 4 characters, also keep
-        #  a lookup of just the first 4 characters
-        if len(word) > 4:
-            WORD_LOOKUP[word[:4]] = i
+    @classmethod
+    def word(cls, index):
```

Comment:
Maybe `get_word(cls, index):` would be less ambiguous?

Reply:
I'll do `__getitem__` so it works more like a dict.